lib/auth/refreshclient.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Auth2Client from './oauth2client';
+import * as util from 'util';
+
+export default class UserRefreshClient extends Auth2Client {
+
+  public _refreshToken: string;
+
+  /**
+   * User Refresh Token credentials.
+   *
+   * @param {string} clientId The authentication client ID.
+   * @param {string} clientSecret The authentication client secret.
+   * @param {string} refreshToken The authentication refresh token.
+   * @constructor
+   */
+  constructor(clientId?: string , clientSecret?: string, refreshToken?: string) {
+    super(clientId, clientSecret);
+    // Named to avoid collision with the method refreshToken_
+    this._refreshToken = refreshToken;
+  }
+
+  // Executes the given callback if it is not null.
+  private callback = (c, err?, res?) => {
+    if (c) {
```

Comment:
I've noticed some areas of the code base use these callback checks while others do `callback || noop`. I know it was copied but it would be nice to have consistency.